test/e2e/obsctl_test.go

```diff
@@ -31,7 +33,22 @@ const (
 // Hydra is spun up externally via start_hydra.sh, as accessing it via docker network is difficult for obsctl.
 // Follows similar pattern as https://observatorium.io/docs/usage/getting-started.md/.
 // Also registers tenants in hydra.
-func preTest(t *testing.T) *e2e.DockerEnvironment {
+
+func hydra() *exec.Cmd {
+
+	cmd := exec.Command("/bin/sh", "start_hydra.sh")
+	if err := cmd.Start(); err != nil {
```

Comment:
Right so `cmd.Start()` runs a separate goroutine which runs in background. This means that the rest of the test progresses in the meanwhile. So the test would be making requests before hydra has even been downloaded.

So we need to wait for the script to finish with `cmd.Wait()`. 🙂